pennylane/transforms/qmc.py

```diff
@@ -93,41 +105,57 @@ def apply_controlled_Q(fn, wires, target_wire, control_wire, work_wires):
             decomposing :math:`\mathcal{Q}`
 
     Returns:
-        function: The input function transformed to the :math:`\mathcal{Q}` unitary
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumTape], function]:
+
+        The transformed circuit as described in :func:`qml.transform <pennylane.transform>`. Executing this circuit
+        will perform control on :math:`\mathcal{Q}` unitary.
 
     Raises:
         ValueError: if ``target_wire`` is not in ``wires``
     """
-    fn_inv = adjoint(fn)
-
-    wires = Wires(wires)
-    target_wire = Wires(target_wire)
-    control_wire = Wires(control_wire)
-    work_wires = Wires(work_wires)
-
-    if not wires.contains_wires(target_wire):
-        raise ValueError("The target wire must be contained within wires")
-
-    @wraps(fn)
-    def wrapper(*args, **kwargs):
-        _apply_controlled_v(target_wire=target_wire, control_wire=control_wire)
-        fn_inv(*args, **kwargs)
-        _apply_controlled_z(wires=wires, control_wire=control_wire, work_wires=work_wires)
-        fn(*args, **kwargs)
-
-        _apply_controlled_v(target_wire=target_wire, control_wire=control_wire)
-        fn_inv(*args, **kwargs)
-        _apply_controlled_z(wires=wires, control_wire=control_wire, work_wires=work_wires)
-        fn(*args, **kwargs)
-
-    return wrapper
-
-
-def quantum_monte_carlo(fn, wires, target_wire, estimation_wires):
+    operations = tape.operations.copy()
+    updated_operations = []
+
+    with qml.queuing.QueuingManager.stop_recording():
+        fn_inv = [adjoint(copy(op)) for op in reversed(operations)]
+
+        wires = Wires(wires)
+        target_wire = Wires(target_wire)
+        control_wire = Wires(control_wire)
+        work_wires = Wires(work_wires)
+
+        if not wires.contains_wires(target_wire):
+            raise ValueError("The target wire must be contained within wires")
+
+        updated_operations.extend(
+            _apply_controlled_v(target_wire=target_wire, control_wire=control_wire)
+        )
+        updated_operations.extend(fn_inv)
+        updated_operations.extend(
+            _apply_controlled_z(wires=wires, control_wire=control_wire, work_wires=work_wires)
+        )
+        updated_operations.extend(operations)
+        updated_operations.extend(
+            _apply_controlled_v(target_wire=target_wire, control_wire=control_wire)
+        )
+        updated_operations.extend(fn_inv)
+        updated_operations.extend(
+            _apply_controlled_z(wires=wires, control_wire=control_wire, work_wires=work_wires)
+        )
+        updated_operations.extend(operations)
+
+    tape = qml.tape.QuantumTape(ops=updated_operations)
```

Comment:
Doesn't seem like we care about the type of the tape, measurements, or shots but I just want to confirm.

Reply:
I added it thanks :+1: